README.md

```diff
@@ -1,33 +1,230 @@
-The service client will update the client.game ENS .
-If after X(5) blocks the reflector.game ENS does not reflect its client.game ENS the client will submit a new claim for the swear contract.
-In this case the claim will include the client.game ENS hash.
-If the client is right a refund + compensation will send back to him from the swindle contract.
+### SwearGame contract
 
-### Phase 2: (Not implemented yet)
-The service will charge X amount of tokens for each reflection operation .
-Client will pay the service before each operation.
--In this case a client claim will include also the payment tx along with the client.game hash.
+This is the main general swear contract which conduct the trial.
 
+It calls the specific TrialRules contract to get the witnesses ,get the trial statuses and transitions
+and to check for expiry time for a specific case and trial status.
 
-## test
-go test ./contracts/ -v --run Test
+It calls the specific game witnesses contracts to testimony and validate for a submitted evidence.
+
+The SwearGame contract is align with the ABIs defined at SwearGameAbstract (courtroomabstract.sol).
+
+/// @notice () - a payable function which is used by the service as a deposit functionality
+
+///The function without name is the default function that is called whenever anyone sends funds to a contract
+
+function () payable;
+
+/// @notice register - register a player to the game
+
+/// The function will throw if the player is already register or there is not
+
+/// enough deposit in the contract to ensure the player could be compensated for the
+
+/// case of a valid case.
+
+/// @param _player  - the player address
+
+/// @return bool registered - true for success registration.
+
+function register(address _player) onlyOwner public returns (bool registered);
+
+/// @notice leaveGame - dismiss a player from the game (unregister)
+
+/// allow only plaintiff which do not have openCases on it name to leave game
+
+/// @param _player  - the player address
+
+function leaveGame(address _player);
+
+/// @notice getStatus - return the trial status of a case
+///
+
+/// @param id  - case id
+
+/// @return  status  - the status of a case
+
+function getStatus(bytes32 id) public constant returns (uint8 status);
+
+/// @notice newCase - open a new case for a service id
+
+/// the function require that the msg sender is already register to the game.
+
+/// @param serviceId  - service id
+
+/// @return bool - true for successful operation.
+
+function newCase(bytes32 serviceId) public returns (bool);
+
+/// @notice trial - initiate or restart a trial process for a certain case
+///
+/// the function require that the case is a valid one.
+
+/// @param id  - case id
+
+/// @return bool - true for successful operation.
+
+function trial(bytes32 id) public returns (bool);
+
+### Expert witness contract
+
+ A smart contract which responsible to testimony for a pre defined
+ evidence submitted to it.
```

Comment:
it means that each specific witness is testimony according to a certain type of evidence which is submitted to it.
for example , the promise validator is asked to testimony according a signed promise which is submitted to it by the client . This signed promise has a specific structure which the promise validator witness familiar with .